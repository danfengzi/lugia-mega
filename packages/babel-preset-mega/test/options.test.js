/**
 * Created Date: Friday, July 19th 2019, 6:42:32 pm
 * Author: hanjingbo@ysstech.com | jingboup@gmail.com
 * -----
 * Last Modified: Monday, July 22nd 2019, 2:48:00 am
 * Modified By: hanjingbo <hanjingbo@ysstech.com | jingboup@gmail.com>
 * -----
 * Copyright (c) 2019-present, #Lugia#.
 * ------------------------------------
 * JavaScript will save your soul!
 */

import { transformFileSync } from '@babel/core';
import { readdirSync } from 'fs';
import { join } from 'path';

function testEngine(engine, env = 'production') {
  it(`engine: ${engine}`, () => {
    const fixtures = join(__dirname, './fixtures');
    const results = readdirSync(fixtures).map(file => {
      const { code } = transformFileSync(join(fixtures, file), {
        presets: [
          [
            require.resolve(join(__dirname, '../src')),
            {
              env,
              engine,
              cwd: join(__dirname, '../')
            }
          ]
        ]
      });
      return {
        file,
        code
      };
    });

    expect(results).toMatchSnapshot();
  });
}

describe('babel-preset-mega', () => {
  // testEngine('webpackApp');
  // testEngine('webpackLib');
  testEngine('nodeApp');
  // testEngine('nodeLib');
});
