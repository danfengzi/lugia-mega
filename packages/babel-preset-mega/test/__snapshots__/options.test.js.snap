// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-preset-mega engine: nodeApp 1`] = `
Array [
  Object {
    "code": "export default (async () => {
  function go() {}

  await go();
});",
    "file": "async.js",
  },
  Object {
    "code": "const set = new Set([1, 2, 3]);
[1, 2, 3].includes(2);",
    "file": "core.js",
  },
  Object {
    "code": "// eslint-disable-next-line
function foo(one, two, three) {}",
    "file": "flow.js",
  },
  Object {
    "code": "// eslint-disable-next-line
import React from 'react'; // eslint-disable-next-line

import ReactDOM from 'react-dom';
ReactDOM.render(React.createElement(\\"h1\\", null, \\"Hello, world!\\"), document.getElementById('root'));",
    "file": "react.jsx",
  },
  Object {
    "code": "// eslint-disable-next-line
import * as React from 'react';

function Hello({
  name,
  enthusiasmLevel = 1,
  onIncrement,
  onDecrement
}) {
  if (enthusiasmLevel <= 0) {
    throw new Error('You could be a little more enthusiastic. :D');
  }

  return React.createElement(\\"div\\", {
    className: \\"hello\\"
  }, React.createElement(\\"div\\", {
    className: \\"greeting\\"
  }, \\"Hello \\", name + getExclamationMarks(enthusiasmLevel)), React.createElement(\\"div\\", null, React.createElement(\\"button\\", {
    type: \\"button\\",
    onClick: onDecrement
  }, \\"-\\"), React.createElement(\\"button\\", {
    type: \\"button\\",
    onClick: onIncrement
  }, \\"+\\")));
}

export default Hello; // helpers

function getExclamationMarks(numChars) {
  return Array(numChars + 1).join('!');
}",
    "file": "react.tsx",
  },
  Object {
    "code": "const x = {
  languageName: '',
  enthusiasmLevel: 1
};
export default x;",
    "file": "typescript.ts",
  },
]
`;
